packages/shared/lib/services/proxy/retry.ts

```diff
@@ -68,22 +68,58 @@ export function getProxyRetryFromErr({ err, proxyConfig }: { err: unknown; proxy
     if (proxyConfig.retryHeader && (proxyConfig.retryHeader.at || proxyConfig.retryHeader.after)) {
         // Headers configured on the fly
         const type = proxyConfig.retryHeader.at ? 'at' : 'after';
-        const retryHeader = proxyConfig.retryHeader.at ? proxyConfig.retryHeader.at : proxyConfig.retryHeader.after;
+        let retryHeaders: string[];
 
-        const res = getRetryFromHeader({ err, type, retryHeader: retryHeader! });
-        if (res.found) {
-            return { retry: true, reason: `custom_${res.reason}`, wait: res.wait };
+        if (proxyConfig.retryHeader.at) {
+            retryHeaders = Array.isArray(proxyConfig.retryHeader.at) ? proxyConfig.retryHeader.at : [proxyConfig.retryHeader.at];
+        } else {
+            retryHeaders = Array.isArray(proxyConfig.retryHeader.after) ? proxyConfig.retryHeader.after : [proxyConfig.retryHeader.after!];
+        }
+
+        // Check all headers and find the most restrictive retry time
+        let bestRetry: { reason: string; wait: number } | null = null;
```

Comment:
I think this is the other way around. We should pick the latest retry (highest wait). If one header says wait 20s, and the other one wait 60s, we should be conservative and pick the 60s. wdyt?

Reply:
I am not sure of the point of having multiple rate limits if the one that matter is the shortest one.
if one says wait 20s and the other one wait 60s, why does the second one exists then if you can retry after 20s

Reply:
I agree with thomas, it's not possible to have two rate limits reset and not use the biggest one; because if the system says that our daily limit is reached, then the minutely limit doesn't matter (unless we are missing something)
If they always send those headers (even if daily limit hasn't been reached), and then we need to check which one is actually relevant with a second header

Reply:
maybe I have no idea how our retry logic works but with the second example (the 500 one) why would the ratelimit reset be accounted for since it is not rate-limited?